interface/drugs/add_edit_drug.php

```diff
@@ -372,263 +368,235 @@ function sel_related() {
+                $pres = sqlStatement("SELECT option_id, title FROM list_options " .
+                    "WHERE list_id = 'taxrate' AND activity = 1 ORDER BY seq");
+                while ($prow = sqlFetchArray($pres)) {
+                    echo "     <td class='font-weight-bold'>" .
+                        generate_display_field(array('data_type' => '1','list_id' => 'taxrate'), $prow['option_id']) .
+                        "</td>\n";
+                }
+                ?>
+            </tr>
+            <?php
+            $blank_lines = $GLOBALS['sell_non_drug_products'] == 2 ? 1 : 3;
+            if ($tres) {
+                while ($trow = sqlFetchArray($tres)) {
+                    $blank_lines = $GLOBALS['sell_non_drug_products'] == 2 ? 0 : 1;
+                    $selector = $trow['selector'];
+                // Get array of prices.
+                    $prices = array();
+                    $pres = sqlStatement(
+                        "SELECT lo.option_id, p.pr_price " .
+                        "FROM list_options AS lo LEFT OUTER JOIN prices AS p ON " .
+                        "p.pr_id = ? AND p.pr_selector = ? AND " .
+                        "p.pr_level = lo.option_id " .
+                        "WHERE lo.list_id = 'pricelevel' AND lo.activity = 1 ORDER BY lo.seq",
+                        array($drug_id, $selector)
+                    );
+                    while ($prow = sqlFetchArray($pres)) {
+                        $prices[$prow['option_id']] = $prow['pr_price'];
+                    }
 
-<?php if (AclMain::aclCheckCore('admin', 'super') && $drug_id) { ?>
-<input class="btn btn-danger" type='submit' name='form_delete' value='<?php echo xla('Delete'); ?>' />
-<?php } ?>
-<input type='button' class="btn btn-secondary" value='<?php echo xla('Cancel'); ?>' onclick='window.close()' />
-</div>
+                    writeTemplateLine(
+                        $selector,
+                        $trow['dosage'],
+                        $trow['period'],
+                        $trow['quantity'],
+                        $trow['refills'],
+                        $prices,
+                        $trow['taxrates']
+                    );
+                }
+            }
 
-</center>
+            for ($i = 0; $i < $blank_lines; ++$i) {
+                $selector = $GLOBALS['sell_non_drug_products'] == 2 ? $row['name'] : '';
+                writeTemplateLine($selector, '', '', '', '', $emptyPrices, '');
+            }
+            ?>
+        </table>
+    </div>
+
+    <div class="btn-group">
+        <button type='submit' class="btn btn-primary btn-save" name='form_save' value='<?php echo  $drug_id ? xla('Update') : xla('Add') ; ?>'><?php echo  $drug_id ? xla('Update') : xla('Add') ; ?></button>
+        <?php if (AclMain::aclCheckCore('admin', 'super') && $drug_id) { ?>
+        <button class="btn btn-danger" type='submit' name='form_delete' value='<?php echo xla('Delete'); ?>'><?php echo xla('Delete'); ?></button>
```

Comment:
Inputs that have the type of "submit" should stay as inputs and not be turned into buttons. This is for form-related reasons. If they have the type of button, then yes you should convert them to a button.

Reply:
Thanks for your explanation, but if we change it into the input, our customer font-awesome icon will fail to show.

Reply:
yep, agree with @stu01509 on this. Submit inputs are now evil since unable to use the cool font-awesome buttons with them: (hint: pseudo-elements: https://github.com/openemr/openemr/issues/3251 )

Reply:
As long as you keep the name and value html attributes, things seem to go well.